### PR TITLE
Move dependency declarations into a version catalog

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -18,16 +18,16 @@ import org.jetbrains.kotlin.gradle.dsl.KotlinProjectExtension
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-  kotlin("jvm") version versions.kotlin apply false
-  id("org.jetbrains.dokka") version versions.dokka apply false
-  id("com.diffplug.spotless") version versions.spotless
-  id("com.vanniktech.maven.publish") version versions.mavenPublish apply false
+  kotlin("jvm") version libs.versions.kotlin.get() apply false
```
Comment (JakeWharton) [inline build.gradle.kts:21]: This DSL is truly horrible

**Review by JakeWharton — APPROVED** [review]
So much nicer! Death to buildSrc!

**Review by ZacSweers — APPROVED** [review]
Nice

File: build.gradle.kts (hunk 1 above)
Comment (Egorand) [reply build.gradle.kts:21]: Indeed, was thinking to look for alternatives

**Comment by Egorand** [comment]
Yeah, I like that there's a 1st class solution now, makes it easier to build tools around it, no need to support all kinds of weird ways people declare dependencies

File: build.gradle.kts (hunk 1 above)
Comment (Goooler) [reply build.gradle.kts:21]: `classpath` thems is a better way ?

Comment (Egorand) [reply build.gradle.kts:21]: @Goooler agreed, this is much nicer! #1130

**Pull request merged**